### PR TITLE
time_manager: returns ds with original time

### LINKED ISSUE #73: calling compute_ann_mean(ds) can lead to ds.time becoming encoded
If I call compute_ann_mean(ds) on an xarray Dataset ds that has
an unencoded time coordinate, and a time bounds named ds.time.attrs['bounds'],
I'm seeing that the time coordinate in ds gets encoded by the compute_ann_mean call.
Changing aspects of ds in compute_ann_mean(ds) is unexpected behavior (to me).
I'm guessing that this is occurring at
https://github.com/NCAR/esmlab/blob/master/esmlab/climatology.py#L209

Below is some jupyter notebook code that demonstrates this.
I'm using the very latest commit of esmlab, be608ed33d89ccff5ec7e194d081b87d0c7df3ad, and
numpy                  1.15.4               
xarray                 0.11.3               

```python
import numpy as np
import xarray as xr
import esmlab
```

```python
# set up values for Dataset, 2 yrs of analytic monthly values
days_1yr = np.array([31.0, 28.0, 31.0, 30.0, 31.0, 30.0, 31.0, 31.0, 30.0, 31.0, 30.0, 31.0])
time_edges = np.insert(np.cumsum(np.concatenate((days_1yr, days_1yr))), 0, 0)
time_bounds_vals = np.stack((time_edges[:-1], time_edges[1:]), axis=1)
time_vals = np.mean(time_bounds_vals, axis=1)
var_vals = np.sin(np.pi * time_vals / 365.0)
```

```python
# create Dataset, including time_bounds
time_var = xr.DataArray(time_vals, name='time', dims='time', coords={'time':time_vals},
                        attrs={'units':'days since 0001-01-01', 'calendar':'noleap',
                               'bounds':'time_bounds'})
time_bounds = xr.DataArray(time_bounds_vals, name='time_bounds', dims=('time', 'd2'),
                           coords={'time':time_var})
var = xr.DataArray(var_vals, name='var', dims='time', coords={'time':time_var})
ds = var.to_dataset()
ds = xr.merge((ds, time_bounds))

print('ds.time before compute_ann_mean')
print(ds.time)
ds_ann = esmlab.climatology.compute_ann_mean(ds)
print('')
print('ds.time after compute_ann_mean')
print(ds.time)
```

    ds.time before compute_ann_mean
    <xarray.DataArray 'time' (time: 24)>
    array([ 15.5,  45. ,  74.5, 105. , 135.5, 166. , 196.5, 227.5, 258. , 288.5,
           319. , 349.5, 380.5, 410. , 439.5, 470. , 500.5, 531. , 561.5, 592.5,
           623. , 653.5, 684. , 714.5])
    Coordinates:
      * time     (time) float64 15.5 45.0 74.5 105.0 ... 623.0 653.5 684.0 714.5
    Attributes:
        units:     days since 0001-01-01
        calendar:  noleap
        bounds:    time_bounds
    
    ds.time after compute_ann_mean
    <xarray.DataArray 'time' (time: 24)>
    array([cftime.DatetimeNoLeap(1, 1, 16, 12, 0, 0, 0, 2, 16),
           cftime.DatetimeNoLeap(1, 2, 15, 0, 0, 0, 0, 4, 46),
           cftime.DatetimeNoLeap(1, 3, 16, 12, 0, 0, 0, 5, 75),
           cftime.DatetimeNoLeap(1, 4, 16, 0, 0, 0, 0, 1, 106),
           cftime.DatetimeNoLeap(1, 5, 16, 12, 0, 0, 0, 3, 136),
           cftime.DatetimeNoLeap(1, 6, 16, 0, 0, 0, 0, 6, 167),
           cftime.DatetimeNoLeap(1, 7, 16, 12, 0, 0, 0, 1, 197),
           cftime.DatetimeNoLeap(1, 8, 16, 12, 0, 0, 0, 4, 228),
           cftime.DatetimeNoLeap(1, 9, 16, 0, 0, 0, 0, 0, 259),
           cftime.DatetimeNoLeap(1, 10, 16, 12, 0, 0, 0, 2, 289),
           cftime.DatetimeNoLeap(1, 11, 16, 0, 0, 0, 0, 5, 320),
           cftime.DatetimeNoLeap(1, 12, 16, 12, 0, 0, 0, 0, 350),
           cftime.DatetimeNoLeap(2, 1, 16, 12, 0, 0, 0, 3, 16),
           cftime.DatetimeNoLeap(2, 2, 15, 0, 0, 0, 0, 5, 46),
           cftime.DatetimeNoLeap(2, 3, 16, 12, 0, 0, 0, 6, 75),
           cftime.DatetimeNoLeap(2, 4, 16, 0, 0, 0, 0, 2, 106),
           cftime.DatetimeNoLeap(2, 5, 16, 12, 0, 0, 0, 4, 136),
           cftime.DatetimeNoLeap(2, 6, 16, 0, 0, 0, 0, 0, 167),
           cftime.DatetimeNoLeap(2, 7, 16, 12, 0, 0, 0, 2, 197),
           cftime.DatetimeNoLeap(2, 8, 16, 12, 0, 0, 0, 5, 228),
           cftime.DatetimeNoLeap(2, 9, 16, 0, 0, 0, 0, 1, 259),
           cftime.DatetimeNoLeap(2, 10, 16, 12, 0, 0, 0, 3, 289),
           cftime.DatetimeNoLeap(2, 11, 16, 0, 0, 0, 0, 6, 320),
           cftime.DatetimeNoLeap(2, 12, 16, 12, 0, 0, 0, 1, 350)], dtype=object)
    Coordinates:
      * time     (time) object 0001-01-16 12:00:00 ... 0002-12-16 12:00:00
    Attributes:
        units:     days since 0001-01-01
        calendar:  noleap
        bounds:    time_bounds

### PR DESCRIPTION
**Pull request by matt-long**
closes #73 

This was not quite as simple as we expected. While `groupby` works on non-coordinates, in practice it is necessary to do the `groupby` on the time coordinate. xarray (via numpy) cannot do math on cftime axes, so these must be the coordinate over which the `groupby` is applied. The solution is to save the old time coordinate, compute on it, then replace it.

@andersy005, this is ready to go.

### CODE REVIEW

File: esmlab/utils/time.py [hunk 1]
```diff
@@ -1,110 +1,222 @@
...
 import numpy as np
 import xarray as xr
 
+from .variables import get_variables
+
+DEFAULT_TIME_UNITS = "days since 0001-01-01 00:00:00"
+DEFAULT_CALENDAR = "standard"
+
+
+class time_manager(object):
```
Comment (andersy005) [inline esmlab/utils/time.py]: This approach is very clean and clear! 👍

**Comment by andersy005** [comment]
Thank you, @matt-long!

File: esmlab/utils/time.py (hunk 1 above)
Comment (andersy005) [reply esmlab/utils/time.py]: Can you add this class to the API page in docs?

**Comment by andersy005** [comment]
One more request: Can you add a file called `codecov.yml` in the root dir of the repo with the following content?:

```yaml
codecov:
  max_report_age: off

comment: false

coverage:
  precision: 2
  round: down
  status:
    project:
      default:
       threshold: 0.2
       if_not_found: success
    patch:
      default:
        enabled: no
        if_not_found: success
    changes:
      default:
        enabled: no
        if_not_found: success
```
This will silence some unnecessary `codecov` checks.

**Comment by matt-long** [comment]
@andersy005, I added doc-strings, but did not include `time_manager` in the api.rst because it's really an internal thing.

**Pull request merged**